### ISSUE TITLE
Using Task in DevOps Pipelines with a DevOps Wiki

### ISSUE THREAD

**Issue by asela-ntt** [rfennell/AzurePipelines#576]
### Running On
Azure DevOps Pipelines

### Version
Latest

### Expected behavior and actual behavior
According to the description I thought this can be used to update a Wiki page in DevOps. However, it is asking for GitName and GitEmail parameters which are mandatory. What should I provide to these when I am using a DevOps wiki?

**Comment by asela-ntt** [comment]
Thanks for your response. I am trying to use your component to update a Wiki page in Azure DevOps.

I used $(Release.RequestedFor) for Git Name and $(Release.RequestedForEmail) for Git Email which picked up my credentials. It did not work.

I ran it in debug mode and this is the error I got.

2019-11-19T23:14:30.7106873Z ##[section]Starting: Git based WIKI Updater
2019-11-19T23:14:30.7236406Z ==============================================================================
2019-11-19T23:14:30.7236469Z Task         : Git based WIKI Updater from Black Marble
2019-11-19T23:14:30.7236513Z Description  : A tools to update or create a WIKI based in a Git repo
2019-11-19T23:14:30.7236539Z Version      : 1.6.2
2019-11-19T23:14:30.7236563Z Author       : Richard Fennell
2019-11-19T23:14:30.7236588Z Help         : Version: 1.6.2. [More Information](https://github.com/rfennell/vNextBuild/wiki/Wiki-Updater-Task/)
2019-11-19T23:14:30.7236631Z ==============================================================================
2019-11-19T23:14:30.9355727Z Variable: Repo [eso-sa.visualstudio.com/TMS/_wiki/wikis/TMS.wiki]
2019-11-19T23:14:30.9355799Z Variable: Filename [ReleaseNoteTest]
2019-11-19T23:14:30.9355878Z Variable: Contents [Release-4]
2019-11-19T23:14:30.9356045Z Variable: Commit Message [Generated By TMS-Sync-Release Pipeline]
2019-11-19T23:14:30.9356198Z Variable: Git Username [Asela Sellahewa]
2019-11-19T23:14:30.9356342Z Variable: Git Email [asela.sellahewa@global.ntt]
2019-11-19T23:14:30.9356465Z Variable: Use Agent Token [true]
2019-11-19T23:14:30.9356609Z Variable: Replace File [false]
2019-11-19T23:14:30.9356715Z Variable: Username [null]
2019-11-19T23:14:30.9356832Z Variable: Password [null]
2019-11-19T23:14:30.9356959Z Variable: Localpath [D:\a\r1\a\repo]
2019-11-19T23:14:30.9357067Z Variable: Data Is File [false]
2019-11-19T23:14:30.9357184Z Variable: SoureFile [null]
2019-11-19T23:14:30.9357304Z Using OAUTH Agent Token, overriding username and password
2019-11-19T23:14:30.9360996Z SYSTEMVSSCONNECTION exists true
2019-11-19T23:14:30.9501315Z URL used https://buildagent:***@eso-sa.visualstudio.com/TMS/_wiki/wikis/TMS.wiki
2019-11-19T23:14:30.9502378Z Cleaned D:\a\r1\a\repo
2019-11-19T23:14:35.1418592Z ##[error]Error: Cloning into 'D:\a\r1\a\repo'...
2019-11-19T23:14:35.1426184Z fatal: unable to update url base from redirection:
2019-11-19T23:14:35.1426524Z   asked for: https://buildagent:***@eso-sa.visualstudio.com/TMS/_wiki/wikis/TMS.wiki/info/refs?service=git-upload-pack
2019-11-19T23:14:35.1426724Z    redirect: https://spsprodeau1.vssps.visualstudio.com/_signin?realm=eso-sa.visualstudio.com&reply_to=https%3A%2F%2Feso-sa.visualstudio.com%2FTMS%2F_wiki%2Fwikis%2FTMS.wiki%2Finfo%2Frefs%3Fservice%3Dgit-upload-pack&redirect=1&hid=2fa95b7a-c774-4595-a6f2-63f482634e9f&context=eyJodCI6MiwiaGlkIjoiZWZhOWIyN2QtZmEzMi00N2MzLWI4ZTctNmQxMmU0OGM2ODNjIiwicXMiOnt9LCJyciI6IiIsInZoIjoiIiwiY3YiOiIiLCJjcyI6IiJ90
2019-11-19T23:14:35.1426751Z
2019-11-19T23:14:35.1427024Z ##[error]Error: Cloning into 'D:\a\r1\a\repo'...
2019-11-19T23:14:35.1427163Z fatal: unable to update url base from redirection:
2019-11-19T23:14:35.1427368Z   asked for: https://buildagent:***@eso-sa.visualstudio.com/TMS/_wiki/wikis/TMS.wiki/info/refs?service=git-upload-pack
2019-11-19T23:14:35.1427446Z    redirect: https://spsprodeau1.vssps.visualstudio.com/_signin?realm=eso-sa.visualstudio.com&reply_to=https%3A%2F%2Feso-sa.visualstudio.com%2FTMS%2F_wiki%2Fwikis%2FTMS.wiki%2Finfo%2Frefs%3Fservice%3Dgit-upload-pack&redirect=1&hid=2fa95b7a-c774-4595-a6f2-63f482634e9f&context=eyJodCI6MiwiaGlkIjoiZWZhOWIyN2QtZmEzMi00N2MzLWI4ZTctNmQxMmU0OGM2ODNjIiwicXMiOnt9LCJyciI6IiIsInZoIjoiIiwiY3YiOiIiLCJjcyI6IiJ90
2019-11-19T23:14:35.1427485Z
2019-11-19T23:14:35.1427748Z fatal: unable to update url base from redirection:
2019-11-19T23:14:35.1427916Z   asked for: https://buildagent:***@eso-sa.visualstudio.com/TMS/_wiki/wikis/TMS.wiki/info/refs?service=git-upload-pack
2019-11-19T23:14:35.1427964Z    redirect: https://spsprodeau1.vssps.visualstudio.com/_signin?realm=eso-sa.visualstudio.com&reply_to=https%3A%2F%2Feso-sa.visualstudio.com%2FTMS%2F_wiki%2Fwikis%2FTMS.wiki%2Finfo%2Frefs%3Fservice%3Dgit-upload-pack&redirect=1&hid=2fa95b7a-c774-4595-a6f2-63f482634e9f&context=eyJodCI6MiwiaGlkIjoiZWZhOWIyN2QtZmEzMi00N2MzLWI4ZTctNmQxMmU0OGM2ODNjIiwicXMiOnt9LCJyciI6IiIsInZoIjoiIiwiY3YiOiIiLCJjcyI6IiJ90
2019-11-19T23:14:35.1429090Z
2019-11-19T23:14:35.1437222Z ##[section]Finishing: Git based WIKI Updater


Settings in my component are given below.

[cid:image001.png@01D59F87.C8E85B90]

Any help is much appreciated. Thanks in advance.

Best Regards,

Asela

**Comment by rfennell** [comment]
The error 'unable to update url base from redirection' is basically a 404 can't find the endpoint

The had a look at the [log for my test case for using the build agent creds for authentication](https://dev.azure.com/richardfennell/GitHub/_releaseProgress?_a=release-environment-logs&releaseId=1544&environmentId=4257)

You can see you have

`URL used https://buildagent:***@eso-sa.visualstudio.com/TMS/_wiki/wikis/TMS.wiki
`
but I have

`URL used https://buildagent:***@dev.azure.com/richardfennell/GitHub/_git/GitHub.wiki`

I think the use is the `/wikis/` as opposed to `/_git/`. I know that 'wikis' is what is shown in the browser when you edit a wiki, but it is not the 'real name'. 

You find this on the WIKI's dropdown menu where there is an option to 'Clone' 

![Capture](https://user-images.githubusercontent.com/324718/69325465-1d780080-0c42-11ea-81d4-f3abfc33932e.PNG)

Does that work for you?

**Comment by asela-ntt** [comment]
Hi Richard,

That helped. Never knew it was a different URL behind the scenes. I clicked clone and found the URL and then pasted it on the browser and hit go. That took me to the git repo and it all made sense after that. Maybe you want to update your documentation with it.

Also I noticed that if you don’t mentioned .md at the file name text box it will create the file without the .md extension in the git repo and it will not show in the wiki. Again another helpful tip to go to the documentation.

Thanks again for all the amazing help.

Best Regards,

Asela
